Домашка за 23.08

DenisProkhorov opened the pull request:
Домашка базируется на задании от 19.08, поэтому комменты именно к DI должны быть общие

- сделал тесты на 2 оговоренных сценария (когда первый раз делаем реквест и когда произвольный раз делаем реквест)
- удалил неиспользуемый логгер
- Открытый вопрос (надо ли было оставлять DI в связке контроллер/сервис, ибо вроде на занятии такого не было)

Commit:
Домашка за 23.08

- сделал тесты на 2 оговоренных сценария (когда первый раз делаем реквест и когда произвольный раз делаем реквест)
- удалил неиспользуемый логгер
- Открытый вопрос (надо ли было оставлять DI в связке контроллер/сервис, ибо вроде на занятии такого не было)

vkhizhin reviewed (commented):
1. Не вижу тестов. Ты видимо забыл их внести в коммит :)
2. Давай разнесем все классы по пакетам. Раньше не писал об этом, но теперь уже думаю пора бы сделать. Все что связано с веб-слоем (контроллеры) в пакет controllers, все что связано с сервис-слоем в пакет service, работа с базой (репозиторий) в пакет repository. LoanRequest и LoanResponse можно вынести в пакет model
3. По поводу вопроса про DI (оставлять ли в связке контроллер/сервис). Не знаю что было на занятии у вас (т.к. лекции нет возможности посещать по времени), но у тебя все правильно сделано, так и должно быть.
4. Я в прошлый раз проглядел видимо. У тебя LoanResponse не используется никак (т.е. он просто есть, но задействован нигде). По-хорошему должно быть так - контроллер должен возвращать LoanResponse. Контроллер получает LoanRequest 
 и прокидывает его дальше в сервис (LoanService). Бизнес логика вся происходит в сервис-слое, например там можно сделать проверку amount у LoanRequest, и если она ниже какого-либо порога вернуть LoanResponse со статусом DENIED, а если все норм, то APPROVED, т.е. выглядит примерно так:
`public LoanResponse createRequest(LoanRequest request){`
`        repo.save(request);`
`        if (request.getAmount() < 10) {`
`            return new LoanResponse(10, ResponseType.DENIED);`
`        }`
`        return new LoanResponse(10, ResponseType.APPROVED);`
`  }`

Commit:
Домашка за 23.08 - заапплаил комменты:
Пункт 1 (тесты) - действительно забыл включить файл с тестами в коммит
Пункт 2 (тесты) - разнес всё по пакетам, как предлагалось
Пункт 3 (вопрос про DI) - все понял, спасибо.
Пункт 4 (логика) - это как раз следующая домашка (сделаю в отдельном коммите)
